### PR TITLE
Add CLI to estimate EMOS coefficients

### PR DESCRIPTION
**Pull request by gavinevans**
Addresses #855 

Description
Add a CLI for estimating EMOS coefficients. This builds on top of the changes in #857. 

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: bin/improver-estimate-emos-coefficients [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    # Regression (NGR)
+    parser.add_argument('distribution', metavar='DISTRIBUTION',
+                        choices=['gaussian', 'truncated gaussian'],
+                        help='The distribution that will be used for '
+                             'calibration. This will be dependent upon the '
+                             'input phenomenon. This has to be supported by '
+                             'the minimisation functions in '
+                             'ContinuousRankedProbabilityScoreMinimisers.')
+    parser.add_argument('cycletime', metavar='CYCLETIME', type=str,
+                        help='The time to be used after at which the '
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply bin/improver-estimate-emos-coefficients]: Done

File: bin/improver-estimate-emos-coefficients [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+       Statistics (EMOS), otherwise known as Non-homogeneous Gaussian
+       Regression (NGR). 2 sources of input data must be provided: historical
+       forecasts and historical truth data (to use in calibration). The
+       estimated coefficients are written to a netCDF file.
+    """
+    parser = ArgParser(
+        description='Estimate coefficients for for Ensemble Model Output '
+                    'Statistics (EMOS), otherwise known as Non-homogeneous '
+                    'Gaussian Regression (NGR)')
+    # Arguments for estimating coefficients for for Ensemble Model Output
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply bin/improver-estimate-emos-coefficients]: Agreed

File: bin/improver-estimate-emos-coefficients [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    parser.add_argument('truth_filepath', metavar='TRUTH_FILEPATH',
+                        help='A path to an input NetCDF file containing the '
+                             'historic truth analyses used for calibration.')
+    parser.add_argument('output_filepath', metavar='OUTPUT_FILEPATH',
+                        help='The output path for the processed NetCDF')
+    # Optional arguments.
+    parser.add_argument('--units', metavar='UNITS',
+                        help='The unit that calibration should be undertaken '
+                             'in. The current forecast, historical forecast '
+                             'and truth will be converted as required.')
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply bin/improver-estimate-emos-coefficients]: Agreed

File: tests/improver-estimate-emos-coefficients/05-predictor_of_mean.bats [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+. $IMPROVER_DIR/tests/lib/utils
+
+@test "estimate-emos-coefficients gaussian realizations history truth output" {
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply tests/improver-estimate-emos-coefficients/05-predictor_of_mean.bats]: Agreed

File: bin/improver-estimate-emos-coefficients [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    parser.add_argument('--predictor_of_mean', metavar='PREDICTOR_OF_MEAN',
+                        choices=['mean', 'realizations'], default='mean',
+                        help='String to specify the input to calculate the '
+                             'calibrated mean. Currently the ensemble mean '
+                             '("mean") and the ensemble realizations '
+                             '("realizations") are supported as the '
+                             'predictors. Default: "mean".')
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply bin/improver-estimate-emos-coefficients]: It probably needs to be reworded to something like `String to specify the predictor used to calibrate the mean`

File: bin/improver-estimate-emos-coefficients (hunk 2 above)
[the remark this answers is not in the document]
Comment (bayliffe) [reply bin/improver-estimate-emos-coefficients]: Done.

File: bin/improver-estimate-emos-coefficients (hunk 3 above)
[the remark this answers is not in the document]
Comment (bayliffe) [reply bin/improver-estimate-emos-coefficients]: Done.

File: bin/improver-estimate-emos-coefficients (hunk 5 above)
[the remark this answers is not in the document]
Comment (bayliffe) [reply bin/improver-estimate-emos-coefficients]: I've gone with that.

File: tests/improver-estimate-emos-coefficients/05-predictor_of_mean.bats (hunk 4 above)
[the remark this answers is not in the document]
Comment (bayliffe) [reply tests/improver-estimate-emos-coefficients/05-predictor_of_mean.bats]: Done.

**Comment by fionaRust** [comment]
@bayliffe  ready for you to merge when you want.

**Pull request merged**